### PR TITLE
Added code to find future warming metrics in manuscript table

### LINKED ISSUE #16: Recreate Table 3 from Hector v3.2.0 Manuscript
Write script to recreate Table 3 from the Hector v3.2.0 manuscript. Before actually writing code, write out in Google Docs pseudocode detailing the following information for each metric:

- What Hector runs are needed to obtain the value and if I need an input file from Kalyn
- Any functions needed and a brief description of them and their arguments/return values/etc.

### PR DESCRIPTION
**Pull request by ptrscll**
I added code to find future warming metrics and included the table outputted by my script. All of my results line up with the data from the manuscript table aside from a couple of entries that are off by 0.01 because of what I assume are rounding errors.

### CODE REVIEW

File: 01/table_making/1F.hector.table.R [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+# args:
+#   data  - data frame outputted by fetchvars
+#   var   - string containing the Hector variable to adjust
+#   start - start year for reference period
+#   end   - end year for reference period
+#
+# returns - data frame containing the normalized values for that variable
+rel_to_interval <- function(data, var, start, end) {
+  benchmark <- get_interval_avg(data, var, start, end)
+  return(rel_to_val(data, var, benchmark))
```
Comment (kdorheim) [inline 01/table_making/1F.hector.table.R:119]: Could you add in a `signif(value, 3)`call before  you return the output? This should take care of rounding differences.

**Review by kdorheim — CHANGES_REQUESTED** [review]
I agree with your comment that the differences are due to rounding/precision differences. Let's see if using the `signif` function can resolve it.

File: 01/table_making/1F.hector.table.R (hunk 1 above)
Comment (ptrscll) [reply 01/table_making/1F.hector.table.R:119]: I just tried adding in signif, but it didn't fix the off-by-0.01 errors.

File: 01/table_making/idealized_inputs/__MACOSX/idealized_inputs/tables/._hector_abruptx4CO2.csv
Comment (kdorheim) [inline 01/table_making/idealized_inputs/__MACOSX/idealized_inputs/tables/._hector_abruptx4CO2.csv:1]: This path is kinda funky, could what does it look like on your side of things?

File: 01/table_making/1F.hector.table.R [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+### Finding TCRE ###
+
+# Setting relevant variables
+tcre_file <- system.file("input/hector_ssp585.ini", package = "hector")
+tcre_vars <- c(GLOBAL_TAS(), FFI_EMISSIONS(), LUC_EMISSIONS())
+tcre_data <- run_hector(ini_file = tcre_file,
+                        yrs = 1750:2300,
+                        vars = tcre_vars)
+
+# Combining emissions data
```
Comment (kdorheim) [inline 01/table_making/1F.hector.table.R:196]: @ptrscll ah okay! so the TCRE calculation comes from looking at the data more like this



total emissions = FFI_EMISSIONS() + LUC_EMISSIONS() - LUC_UPTAKE() -  DACCS_UPTAKE() and to get the cumulative sum of CO2 emissions use the function `cumsum`

File: 01/table_making/1F.hector.table.R [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+                       yrs      = 1800:2070,
+                       vars     = c(GLOBAL_TAS(), CONCENTRATIONS_CO2()))
+
+tcr_co2   <- filter(tcr_data, variable == CONCENTRATIONS_CO2())$value
+tcr_temps <- filter(tcr_data, variable == GLOBAL_TAS())$value
+
+# Finding TCR from linear fit
+tcr_reg <- lm(tcr_temps ~ tcr_co2)
+
+initial_co2 <-
```
Comment (kdorheim) [inline 01/table_making/1F.hector.table.R]: Just a heads up the `PREINDUSTRIAL_CO2()` function can be used with `fetchvars` to get this info and I think that might be where our differences are coming from.

File: 01/table_making/1F.hector.table.R [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+#   yrs  - vector of years to sum these variables for
+#
+# returns: original data frame with additional rows containing the sums of the
+#          values for the inputted variables
+#
+# Note: Currently, all columns other than year, variable, and value will be set
+#       equal to the same values as the last row in the original data frame
+sum_vars <- function(data, vars, name, unit, yrs) {
+
+  # Setting up iterator to iterate through new rows of data frame
```
Comment (kdorheim) [inline 01/table_making/1F.hector.table.R:144]: this is impressive! but the `dplyr` package supports some helpful functions like `group_by()` that do similar things

**Review by kdorheim — CHANGES_REQUESTED** [review]

File: 01/table_making/1F.hector.table.R (hunk 3 above)
Comment (ptrscll) [reply 01/table_making/1F.hector.table.R]: I tried using PREINDUSTRIAL_CO2() but that didn't change my TCR estimate (see latest commit)

File: 01/table_making/1F.hector.table.R (hunk 2 above)
Comment (ptrscll) [reply 01/table_making/1F.hector.table.R:196]: This worked!

**Pull request merged**